src/coingecko_api/base_client.py

```
@@ -0,0 +1,97 @@
+@dataclass
+class BaseAPIConfig:
+    """Configuration for the CoinGecko API client."""
+    base_url: str = "https://api.coingecko.com/api/v3"
+    timeout: int = 10
+    retries: int = 3
+    backoff_factor: float = 0.5
+
+class CoinGeckoBaseClient:
+    """Base client for interacting with the CoinGecko API."""
+
+    def __init__(self, config: BaseAPIConfig = BaseAPIConfig()):
```

Ruff (0.11.9)

Do not perform function call `BaseAPIConfig` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)